src/transport/grpc/mod.rs

```diff
@@ -73,11 +79,30 @@ pub enum GrpcTransportError {
     MessageHubProtoConversion(#[from] MessageHubProtoError),
 }
 
+pub(crate) struct NodeIdInterceptor(Uuid);
+
+impl NodeIdInterceptor {
+    const NODE_ID_METADATA_KEY: &'static str = "node_id";
+
+    pub(crate) fn new(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
+}
+
+impl Interceptor for NodeIdInterceptor {
+    fn call(&mut self, mut req: Request<()>) -> Result<Request<()>, Status> {
+        let val = MetadataValue::from_str(&self.0.to_string())
+            .map_err(|_| Status::new(Code::InvalidArgument, "invalid metadata value"))?;
+        req.metadata_mut().insert(Self::NODE_ID_METADATA_KEY, val);
+        Ok(req)
+    }
+}
+
 /// Shared data of the grpc connection, this struct is internal to the [`Grpc`] connection
 /// where is wrapped in an arc to share an immutable reference across tasks.
 #[derive(Debug)]
 pub struct SharedGrpc {
-    client: Mutex<MessageHubClient<tonic::transport::Channel>>,
+    client: Mutex<MessageHubClient<InterceptedService<Channel, NodeIdInterceptor>>>,
```

Comment:
Add a type alias to simplify the message hub client, in all the places it's used.